src/core/agent/services/identifierService.ts

```diff
@@ -282,16 +282,82 @@ class IdentifierService extends AgentService {
     );
     if (unSyncedData.length) {
       //sync the storage with the signify data
+      let groupId;
       for (const identifier of unSyncedData) {
-        await this.identifierStorage.createIdentifierMetadataRecord({
-          id: identifier.prefix,
-          displayName: identifier.prefix, //same as the id at the moment
-          theme: 0,
-        });
+        const checkDeletedIdentifier = this.checkIdentifierNameIsDeleted(identifier.name);
+        if(checkDeletedIdentifier) continue;
+        const checkNameIsMultiSig = this.checkIdentifierNameIsMultiSig(identifier.name);
+        if(checkNameIsMultiSig){
+          groupId = identifier.name.split(":")[1];
+          const groupMetadata = {
+            groupId,
+            groupCreated: false,
+            groupInitiator: this.checkGroupInitiator(groupId)
+          }
+          await this.identifierStorage.createIdentifierMetadataRecord({
+            id: identifier.prefix,
+            displayName: groupId,
+            theme: 0,
+            groupMetadata
+          });
+        }
+        else {
+          if(identifier.group){
+            const multisigManageAid = identifier.group.mhab.prefix;
+            groupId = identifier.group.mhab.name.split(":")[1];
+            const op = await this.props.signifyClient.operations().get(`group.${identifier.prefix}`)
+            const isPending = !op.done;
+            if(isPending){
+              const pendingOperation = await this.operationPendingStorage.save({
+                id: op.name,
+                recordType: OperationPendingRecordType.Group,
+              });
+              this.props.eventEmitter.emit<OperationAddedEvent>({
+                type: EventTypes.OperationAdded,
+                payload: { operation: pendingOperation },
+              });
+            }else{
+              await this,this.identifierStorage.updateIdentifierMetadata(multisigManageAid,{
```

Comment:
Also, this is completely wrong. If the GROUP identifier is PENDING, then it's because the group has not finished completing yet, which means it might be missing signatures or witness signatures etc.

If the group identifier EXISTS, pending OR not, the group has been created.

Please take the time to actually remember the difference between `groupMetadata` and `multisigManageAid`.